web/modules/custom/joinup_seo/joinup_seo.behat.inc

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/**
+ * @file
+ * Contains \JoinupSeoSubContext.
+ */
+
+declare(strict_types = 1);
+
+use Behat\Gherkin\Node\TableNode;
+use Drupal\DrupalExtension\Context\DrupalSubContextBase;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+use PHPUnit\Framework\Assert;
+
+/**
+ * Behat step definitions for testing SEO.
+ */
+class JoinupSeoSubContext extends DrupalSubContextBase implements DrupalSubContextInterface {
+
+  /**
+   * Assert a meta tag value in the page.
+   *
+   * @param string $meta_name
+   *   The meta tag name.
+   * @param string $meta_value
+   *   The meta tag value.
+   *
+   * @throws \Exception
+   *   Thrown when an tag is not found or the value is not correct.
+   *
+   * @Given the :meta_name metatag should be set to :meta_value
+   */
+  public function assertMetatagInPage(string $meta_name, string $meta_value): void {
+    $xpath = "//head/meta[@name=\"{$meta_name}\" and @content=\"{$meta_value}\"]";
+    if (empty($this->getSession()->getPage()->find('xpath', $xpath))) {
+      throw new Exception("The meta property '{$meta_name}' was either not found or the value is not set to '{$meta_value}'");
+    }
+  }
+
+  /**
+   * Asserts that the schema.org metatags are attached in page.
+   *
+   * @Given the metatag json should be attached in the page
+   */
+  public function assertJsonMetatagsInPage(): void {
+    $json = $this->getMetatagsAsJson();
+    Assert::assertNotEmpty($json, 'Entity metadata are found in the table.');
+    // Assert that the context of the metatag is 'schema.org' to ensure that the
+    // correct metatag is loaded and exists.
+    Assert::assertArrayHasKey('@context', $json, '@context key not found in the metatag array.');
+    Assert::assertEquals($json['@context'], 'https://schema.org', 'The @context property value is not set to the appropriate url.');
+  }
+
+  /**
+   * Asserts that the schema.org metatags are attached in page.
+   *
+   * @Given the metatag json should not be attached in the page
+   */
+  public function assertMetatagsNotInPage(): void {
+    $json = $this->getMetatagsAsJson();
+    Assert::assertEmpty($json, 'Entity metadata are not attached in the page.');
+  }
+
+  /**
+   * Assert the amount of entity graphs of a certain type.
+   *
+   * @param string $count
+   *   The number of graphs of the given type.
+   * @param string $type
+   *   The type of graph which is the schema class.
+   *
+   * @Given :count metatag graph of type :type should exist in the page
+   */
+  public function assertNumberOfEntityGraphsExist(string $count, string $type): void {
```

Comment:
`$count` should be an `int`.

Reply:
I did not know that it automatically is converted into an integer.